src/main/java/com/connexta/search/index/controllers/IndexController.java

```diff
@@ -47,7 +49,18 @@
               ACCEPT_VERSION_HEADER_NAME, acceptVersion, expectedAcceptVersion));
     }
 
-    indexService.index(datasetId, indexRequest.getIrmLocation());
+    // TODO: Change all instances of URI to URL. This is just a temporary workaround for now to work
+    // with the current version of the Index API.
+    URI irmLocation;
+    try {
+      irmLocation = new URI(indexRequest.getIrmLocation());
+    } catch (URISyntaxException e) {
+      throw new DetailedResponseStatusException(
```

Comment:
❓ Should we be throwing something custom like an `IndexException`?

Reply:
It was decided that we will just leave this as-is for now since we will be fixing exception handling in another ticket.